### PR TITLE
(EAI-137) [Q&A] Custom request origins

### PR DESCRIPTION
**Pull request by nlarew**
Jira: (EAI-137) [Q&A] Custom request origins

## Changes

- Adds support for `X-Request-Origin` (with fallback to `Origin`) on the server
- Updates client to automatically set `X-Request-Origin` to the full page URL

Screenshot 2023-11-08 at 5 00 16 PM

### CODE REVIEW

File: chat-server/src/middleware/requestOrigin.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { Request, Response, NextFunction } from "express";
 import { getRequestId, logRequest, sendErrorResponse } from "../utils";
 
+export const CUSTOM_REQUEST_ORIGIN_HEADER = "x-request-origin";
```
Comment (mongodben) [inline chat-server/src/middleware/requestOrigin.ts:4]: should this be Title-Case, `X-Request-Origin`?

Comment (nlarew) [reply chat-server/src/middleware/requestOrigin.ts:4]: Doesn't really matter, headers are case-insensitive

File: chat-server/src/routes/conversations/createConversation.test.ts [hunk 2]
```diff
@@ -35,8 +35,11 @@ describe("POST /conversations", () => {
     expect(count).toBe(1);
   });
 
-  it("should respond 400 if the Origin header is missing", async () => {
+  it("should respond 400 if neither Origin nor X-Request-Origin is present", async () => {
```
Comment (mongodben) [inline chat-server/src/routes/conversations/createConversation.test.ts:38]:
```suggestion
  it("should respond 400 if neither Origin nor X-Request-Origin header is present", async () => {
```

File: chat-ui/src/services/conversations.ts [hunk 3]
```diff
@@ -26,6 +26,15 @@ export function formatReferences(references: References): string {
   return [heading, ...listOfLinks].join("\n\n");
 }
 
+export const CUSTOM_REQUEST_ORIGIN_HEADER = "x-request-origin";
```
Comment (mongodben) [inline chat-ui/src/services/conversations.ts:29]:
```suggestion
export const CUSTOM_REQUEST_ORIGIN_HEADER = "X-Request-Origin";
```

for consistency w above

**Review by mongodben — APPROVED** [review]
looks good. added a few nitty suggestions

File: chat-ui/src/services/conversations.ts (hunk 3 above)
Comment (nlarew) [reply chat-ui/src/services/conversations.ts:29]: This is a good example of something that would make sense in `chat-core` but not `rag-core`

**Pull request merged**